src/shared/nested-dropdown-list/nested-dropdown-list-helper.ts

```diff
@@ -36,6 +88,40 @@ export namespace NestedDropdownListHelper {
         return list;
     };
 
+    export const updateSelectedAll = <V1, V2, V3>(
+        currentItems: FormattedOptionMap<V1, V2, V3>,
+        isSelectedAll: boolean
+    ): UpdateSelectedAllType<V1, V2, V3> => {
+        let list = currentItems;
+        let keyPaths: string[][] = [];
+        let items: SelectedItem<V1, V2, V3>[] = [];
+
+        if (isSelectedAll) {
+            const { keyPaths: _keyPaths, items: _items } =
+                getAllKeyPathsAndItems(list);
+
+            keyPaths = _keyPaths;
+            items = _items;
+
+            list = produce(
+                currentItems,
+                (draft: FormattedOptionMap<V1, V2, V3>) => {
+                    keyPaths.forEach((key) => {
+                        const parentKey = key.slice(0, -1);
+                        const item = getItemAtKeyPath(draft, parentKey);
+                        item.expanded = true;
```

Comment:
L1 item does not get expanded if it does not have a non-category L2 item, is that intended?

Reply:
it expanded. i use below `options` please let me know if i misunderstand.

```
[
    {
        label: "Category 1",
        value: { id: 999, name: "category 1" },
        key: "999",
        subItems: [
            {
                label: "Option 1",
                value: {
                    id: 10001,
                    name: "Option-1",
                },
                key: "10001",
            },
            {
                label: "Sub Category A",
                value: {
                    id: 820,
                    name: "Sub category a",
                },
                key: "820",
            },
        ],
    },
    {
        label: "Hive",
        value: { id: 487, name: "hive" },
        key: "32",
        subItems: [
            {
                label: "Level 8",
                value: { id: 100, name: "Level 8" },
                key: "100",
            },
            {
                label: "Level 9",
                value: { id: 101, name: "Level 9" },
                key: "101",
            },
            {
                label: "Level 10",
                value: { id: 102, name: "Level 10" },
                key: "102",
            },
        ],
    },
];
``

Reply:
the L1 is `expanded: true`, but L2 which non-category `expanded: true` as well. 
I'll fix this.

Reply:
> may i have the options? i tested, it expanded.

with the current options,
1. collapse Category 1
2. click Select All

Hive and the subcategories are expanded, but not Category 1

Reply:
interesting. i didn't expected that... let me check. thanks pointed this out.